### PR TITLE
Fix all non-data-loss warnings in Visual Studio

### PR DESCRIPTION
**Pull request by lemire**
Let see how bad things get with visual studio when we do not suppress any warning.

### CODE REVIEW

**Comment by lemire** [comment]
cc @jkeiser 

I think we can just move the deprecated definitions inline to avoid trouble.

**Comment by jkeiser** [comment]
@lemire I fixed one set of them in #752. Needs some work though, as it causes a perf regression the way I did it.

**Comment by lemire** [comment]
@jkeiser Ok. Let us see if this PR of mine can fix all the non-conversion errors. That would be a step forward.

File: include/simdjson/parsedjson_iterator.h [hunk 1]
```diff
@@ -15,12 +15,47 @@
 #include "simdjson/internal/jsonformatutils.h"
 
 namespace simdjson {
+// We moved the definitions of the ctor inline for this deprecated
+// class to avoid spurious warnings under Visual Studio.
+
 
 class [[deprecated("Use the new DOM navigation API instead (see doc/usage.md)")]] dom::parser::Iterator {
 public:
-  inline Iterator(const dom::parser &parser) noexcept(false);
```
Comment (jkeiser) [inline include/simdjson/parsedjson_iterator.h:21]: Clever way to fix, and seems totally fine to me!

Comment (lemire) [reply include/simdjson/parsedjson_iterator.h:21]: I feel better about this way of doing things that the removal of deprecated under Visual Studio. I feel like Visual Studio should be a first citizen, as much as possible, with top-notch support...

Comment (lemire) [reply include/simdjson/parsedjson_iterator.h:21]: The price we pay is that we make the code a bit uglier, but given that this is deprecated code...

Comment (jkeiser) [reply include/simdjson/parsedjson_iterator.h:21]: I tried fixing it by moving the [[deprecated]] to the definition (which is in dom::parser). But I agree, it's deprecated code and I don't want to futz with it.

**Comment by lemire** [comment]
@jkeiser I expect that the flags will go green. I'd like for us to merge this. I fixes part of the problem with Visual Studio.

**Comment by jkeiser** [comment]
@lemire VS2017 is angry still. Let me try a solution with deprecation warnings disabled in that file.

**Comment by jkeiser** [comment]
@lemire let's see how https://ci.appveyor.com/project/lemire/simdjson-jmmti/builds/32316118 goes ...

**Comment by jkeiser** [comment]
@lemire heh, that was an attempt at being subtle, but [this](https://github.com/simdjson/simdjson/pull/755) hits the problem with a hammer. Hopefully it stays dead.

**Comment by lemire** [comment]
I am going to close this PR.

**Pull request closed without merging**